.github/workflows/tests.yml

```diff
@@ -68,27 +68,33 @@ jobs:
   tests-canary:
     if: ${{ !contains(github.event.head_commit.message, '[skip main]') }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java-version: [ "11", "17", "21" ]
     steps:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Setup Java
         uses: actions/setup-java@v3
         with:
           distribution: "corretto"
-          java-version: 17
+          java-version: ${{ matrix.java-version }}
       - name: Tests
         run: ./gradlew canary:test canary:build
   tests-server:
     if: ${{ !contains(github.event.head_commit.message, '[skip main]') }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java-version: [ "11", "17", "21" ]
     steps:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Setup Java
         uses: actions/setup-java@v3
         with:
           distribution: "corretto"
-          java-version: 17
+          java-version: ${{ matrix.java-version }}
```

Comment:
Do we really need to run a matrix for the server? I think we can just support only the latest version of Java SDK, especially since we will be distributing it as a docker image in production.

For the `sdk-java`, a matrix is a very good idea.

Reply:
I agree, server just the main version, sdk-java down to 17.

Reply:
The latest LTS is java 21, so far let's have a matrix with 17 and 21, and then let's remove 17

Reply:
sdk-java should continue to support Java 11. Kafka supports all the way down to Java 8. Some of our users may be really outdated; I would hate to lose a deal because we had one less step in our pipeline